### PR TITLE
Integrate giving a review with the API (#79)

### PR DESCRIPTION
**Pull request by damdiana**

### CODE REVIEW

File: src/components/ProductReviewsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  initialReviews: ProductReviewsSection[];
+}) => {
+  const [productReviews, setProductReviews] =
+    useState<ProductReview[]>(initialReviews);
+
+  async function onAddReview(review: ProductReviewPostPayload) {
+    let resp = await postProductReview(review);
+    if (resp.ok) {
+      setProductReviews([...productReviews, resp.productReview]);
+    }
```
[the remark this answers is not in the document]
Comment (iampava) [reply src/components/ProductReviewsSection.tsx:30]: Actually no need for try-catch here. We have one inside `<FeedbackForm.tsx>`. Just make sure to change the state there

File: src/components/FeedbackForm/FeedbackForm.tsx [hunk 2]
```diff
@@ -4,59 +4,110 @@ import { faStar as faStarOutline } from "@fortawesome/free-regular-svg-icons";
...
+              key={index}
+              onClick={() => {
+                setRating(index + 1);
+              }}
+            >
+              <FontAwesomeIcon
+                icon={rating >= index + 1 ? faStar : faStarOutline}
+              />
+            </button>
+          ))}
```
Comment (iampava) [inline src/components/FeedbackForm/FeedbackForm.tsx:67]: This should be extracted into a separate component

```tsx
const RatingButtons = ({
  rating,
  onRatingChange
}: {
  rating: number;
  onRatingChange: (newRating: number) => void;
}) => {
  // ...
}
```

**Review by iampava — CHANGES_REQUESTED** [review]

File: src/app/api/product-reviews/[productId]/route.ts [hunk 3]
```diff
@@ -8,6 +9,17 @@ export async function GET(
...
+    if (product === undefined) {
+      return NextResponse.json(
+        {
+          message: "There is no product with this ID",
+        },
+        {
+          status: 400,
+        }
+      );
+    }
```
[the remark this answers is not in the document]
Comment (damdiana) [reply src/app/api/product-reviews/[productId]/route.ts:22]: will extract this function after this PR is merged

File: src/components/FeedbackForm/FeedbackForm.tsx [hunk 4]
```diff
@@ -4,59 +4,105 @@ import { faStar as faStarOutline } from "@fortawesome/free-regular-svg-icons";
...
+        rating: rating,
+      });
+    } catch (err) {
+      setFormError("Failed to add review");
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // const onRatingChange = () => {
```
Comment (iampava) [inline src/components/FeedbackForm/FeedbackForm.tsx:47]: Why do we have this commented function? Delete if we don't need it

**Review by iampava — APPROVED** [review]
After you remove that commented code, make sure to test everything works. If yes, then merge

**Pull request merged**